### ISSUE TITLE
Error Unnecessary enabling

### ISSUE THREAD

**Issue by Magikdidi24** [rubocop/rubocop#14717]
*Be clear, concise and precise in your description of the problem.
Open an issue with a descriptive title and a summary in grammatically correct,
complete sentences.*

*Use the template below when reporting bugs. Please, make sure that
you're running the latest stable RuboCop and that the problem you're reporting
hasn't been reported (and potentially fixed) already.*

*Before filing the ticket you should replace all text above the horizontal
rule with your own words.*

--------

## Steps to reproduce the problem

```ruby
# rubocop:push -Layout/SpaceAroundMethodCallOperator
self.stage =
  if 'macro' .start_with? stage; Test::MACRO
  end
# rubocop:pop
```

## Expected behavior

No offense should be reported. The rubocop:push -CopName directive temporarily disables a cop, it does not "enable" anything.

## Actual behavior

Unnecessary enabling of -Layout/SpaceAroundMethodCallOperator. (error:Lint/RedundantCopEnableDirective)

## Root cause
The extra_enabled_comments_with_names method in lib/rubocop/comment_config.rb does not filter out push/pop directives. When it encounters # rubocop:push -CopName, the -CopName part is passed to handle_switch, which incorrectly interprets it as an enable directive.

## Proposed fix
In lib/rubocop/comment_config.rb, add a guard clause in extra_enabled_comments_with_names to skip push and pop directives:

```ruby
def extra_enabled_comments_with_names(extras:, names:)
  each_directive do |directive|
    next unless comment_only_line?(directive.line_number)
     -- next if directive.push? || directive.pop? -- 

    if directive.enabled_all?
      handle_enable_all(directive, names, extras)
    else
      handle_switch(directive, names, extras)
    end
  end

  extras
end
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ [bundle exec] rubocop -V
1.82.0 (using Parser 3.3.5.0, rubocop-ast 1.32.3, analyzing as Ruby 3.3, running on ruby 3.3.5) [x86_64-linux]
  - rubocop-performance 1.22.1
  - rubocop-rspec 3.1.0
```
